src/styles/_variables.scss

```diff
@@ -27,10 +27,67 @@ $metadata-modal-breakpoint: 700px;
 $global-toolbar-width: 80px;
 $run-list-controls-height: 95px;
 
-// -- levels -- //
-$z-index-metadata-panel: 3;
-$z-index-metadata-code: 3;
-$z-index-MuiTreeItem-icon: 1;
+// z-index values
+$base: 0;
+$above: 1;
+$below: -1;
+
+$zindex-metadata-code: $base + ($above * 3);
+
+$zindex-MuiTreeItem-icon: $base + $above;
+
+$zindex-sidebar: $base + $above;
+$zindex-ui: $base + ($above * 2);
+$zindex-toolbar: $base + ($above * 2);
+$zindex-sidebar-expanded: $base + ($above * 3);
+
+$zindex-plot-modal: $base + ($above * 3);
+
+$zindex-sticky-elements: $base + $above;
+
+$zindex-search-icon: $base + $above;
+
+$zindex-feature-hints: $base + ($above * 5);
+
+$zindex-modal: $base + ($above * 6);
+
+$zindex-go-back-btn: $base + ($above * 3);
+
+$zindex-global-toolbar: $base + ($above * 5);
+
+$zindex-toolbar-label: $base + $above;
+
+$zindex-metadata-modal: $base + ($above * 6);
+
+$zindex-metadata-panel: $base + ($above * 3);
+$zindex-metadata-link: $base + $above;
+
+$zindex-nodelist-placeholder: $base + ($above * 2);
+$zindex-nodelist-heading: $base + $above;
+
+$zindex-group-background-fade: $base + $below;
+
+$zindex-switch-root: $base;
+$zindex-switch-base: $base + $above;
```

Comment:
hey this is a good start, i think the idea is to set something as a base, in this case 
`$zindex-switch-root: $base;`

then for `$zindex-switch-base: $above + $zindex-switch-root`
and `$zindex-switch-track: $below + $zindex-switch-root`